binderhub/repoproviders.py

```diff
@@ -117,16 +117,27 @@ def _default_auth(self):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        spec_parts = self.spec.split('/')
+        self.user, self.repo, self.unresolved_ref = self.process_spec(self.spec)
+        self.repo = strip_suffix(self.repo, ".git")
+
+    @staticmethod
+    def process_spec(spec):
```

Comment:
Two thoughts: we could make these functions in the file instead of @staticmethod'ing them? And naming: `process_spec` makes me think "you mean a function that processes its argument? Crazy!" :) Maybe name it `_tokenize` or `_split`? Not sure `_tokenize` is much better :-/

Reply:
@betatim I'm glad you raise this, as I had thought about this for a bit too. My thought is that `strip_suffix` definitely can be disconnected, but the `tokenize_spec` is tied to the way Github does things (I think, granted I haven't checked other VCS providers), so I would lean towards keeping that as a `staticmethod`.

I'll start by moving `strip_suffix` outside the class, and can move the other too if you feel strongly :)

Reply:
Also, appreciate the constructive critique about naming

Reply:
We could reuse the spec-splitting for GitLab I reckon, if that makes a difference to you ;)